test/common/testinfra.py

```diff
@@ -565,13 +568,12 @@ def scan_for_pull_tasks(self, update, context, except_context=False):
                         contexts.pop(ctx)
 
         results = []
-        master_contexts = []
-        pull_contexts = []
-        for (context, what) in contexts.items():
-            if "master" in what:
-                master_contexts.append(context)
-            if "pulls" in what:
-                pull_contexts.append(context)
+        branch_contexts = { }
+        for (context, branches) in contexts.items():
+            for branch in branches:
+                if branch not in branch_contexts:
+                    branch_contexts[branch] = [ ]
+                branch_contexts[branch].append(context)
```

Comment:
Hint for the future: The "create initial value entry if it doesn't exist" pattern is usually written as

    branch_contexts.setdefault(branch, []).append(contexts)